### ISSUE TITLE
Not able to use FP16 in pytorch-pretrained-BERT

### ISSUE THREAD

**Issue by Ashish-Gupta03** [huggingface/pytorch-pretrained-BERT#139]
I'm not able to work with FP16 for pytorch BERT code. Particularly for BertForSequenceClassification, which I tried and got the issue
**Runtime error: Expected scalar type object Half but got scalar type Float for argument #2 target**
when I enabled fp16.
Also when using 
`logits = logits.half()
labels = labels.half()`
then the epoch time also increased.

_Originally posted by @Ashish-Gupta03 in https://github.com/huggingface/pytorch-pretrained-BERT/issue_comments#issuecomment-449096213_
